pkg/action/upgrade_test.go

```diff
@@ -142,6 +151,34 @@ var _ = Describe("Runtime Actions", func() {
 		AfterEach(func() {
 			ghwTest.Clean()
 		})
+		It("calculates the recovery source size correctly", func() {
+			dummySourceFile = createDummyFile(fs, dummySourceSizeMb)
+			upgradeConfig := agent.ExtraConfigUpgrade{}
+			upgradeConfig.Upgrade.Entry = constants.BootEntryRecovery
+			upgradeConfig.Upgrade.RecoverySystem.URI = fmt.Sprintf("file:%s", dummySourceFile)
+			d, err := json.Marshal(upgradeConfig)
+			Expect(err).ToNot(HaveOccurred())
+			cliConfig := string(d)
+
+			config, err := agentConfig.Scan(collector.Directories(), collector.Readers(strings.NewReader(cliConfig)))
```

Comment:
why are you passing the default dirs? It would be saner for the test to just pass the only config that we want no?

Reply:
Good point, fixing.